lib/modules/platform/bitbucket-server/index.ts

```diff
@@ -340,9 +340,10 @@ export async function getPr(
     return null;
   }
 
-  const opts: HttpOptions = refreshCache
-    ? { memCache: false }
-    : { cacheProvider: memCacheProvider };
+  const opts: HttpOptions = { memCache: false };
+  if (!refreshCache) {
+    opts.cacheProvider = memCacheProvider;
+  }
```

Comment:
Can this be documented? Both before and after, the intent of this is a little confusing. 

Before: if refreshCache then memCache=false, else cacheProvider=memCacheProvider
After: if refreshCache then memCache=false, else memCache=false and cacheProvider=memCacheProvider

Reply:
Do you mean documented by a comment? I agree, one sec.

Reply:
How do you like it?

```ts
    // Replaces memCache with memCacheProvider. Note that memCache needs to be
    // explicitly disabled as it's enabled by default.
    const opts: HttpOptions = { memCache: false };
    // refreshCache means not to read this request from cache or save it to
    // cache, thus, when refreshCache is true, cacheProvider is not set.
    if (!refreshCache) {
      opts.cacheProvider = memCacheProvider;
    }
```

Reply:
Actually there is one discrepancy with the old (prior to memCacheProvider) behavior and now.

Previously, even if memCache was disabled, the cache would still be saved. I.e. it would disregard existing cache but would update it.

How do we retain the same behavior with memCacheProvider?

Right now when cacheProvider is unset, the response will not be saved to cache. And otherwise it would both be read from cache and saved to cache.

Reply:
Sorry, I got things wrong. The previous behavior was not as I described.

The option `refreshCache` was poorly implemented though, at least in Gerrit and BitBucket, given they were not actually refreshing cache, but only ignoring it.

Reply:
Please see https://github.com/renovatebot/renovate/pull/35056/commits/f6ebf374e3690b004007a682ab22a58a3e0c757b